### PR TITLE
Extract Header menu from Auth to commons

### PR DESCRIPTION
**Pull request by Kamahl19**
#150

### CODE REVIEW

File: src/common/components/Navbar/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import PropTypes from 'prop-types';
+
+import ResponsiveNavigation from '../../../packages/responsive-navigation';
+
+import { Icon, Popover, Menu } from '../';
+
+const Navbar = ({ children }) => (
+  <ResponsiveNavigation>
+    {({ activePath, isMobile, hideNavigation, showNavigation, isNavigationVisible }) => {
+      const Menu = cloneElement(children, { activePath, isMobile });
```
Comment (MiroslavPetrik) [inline src/common/components/Navbar/index.js]: Shouldn't we use `React.Children.only(children)` here?

Or the `cloneElement` clones all the children while injecting the props to them?

File: src/features/auth/components/PageLayout/index.js [hunk 2]
```diff
@@ -3,35 +3,32 @@ import PropTypes from 'prop-types';
...
-              visible={isNavigationVisible}
-            >
-              <Icon type="bars" style={{ color: '#fff' }} />
-            </Popover>
-          ) : (
-            <Navigation activePath={activePath} />
-          )
-        }
-      </ResponsiveNavigation>
+      <Navbar>
```
Comment (MiroslavPetrik) [inline src/features/auth/components/PageLayout/index.js]: Nice & tidy client code with this Navbar! Bravo

File: src/features/auth/components/PageLayout/index.js [hunk 3]
```diff
@@ -3,35 +3,32 @@ import PropTypes from 'prop-types';
...
+              <Trans i18nKey="nav.login">Log In</Trans>
+            </Link>
+          </Navbar.MenuItem>
+        </Navbar.Menu>
+      </Navbar>
     </Layout.Header>
     <Layout.Content>{children}</Layout.Content>
-    <Footer />
+    <Layout.Footer>
+      <Trans i18nKey="footer.copyright">Copyright</Trans>
```
Comment (MiroslavPetrik) [inline src/features/auth/components/PageLayout/index.js]: Do we need footer?

File: src/common/components/Navbar/index.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+      return isMobile ? (
+        <Popover
+          content={Menu}
+          onVisibleChange={visible => (visible ? showNavigation() : hideNavigation())}
+          title={<Icon type="close" onClick={hideNavigation} />}
+          trigger="click"
+          visible={isNavigationVisible}
+        >
+          <Icon type="bars" style={{ color: '#fff' }} />
```
Comment (MiroslavPetrik) [inline src/common/components/Navbar/index.js]: Maybe make the icon a default prop.
The default is a bit ugly, in most cases you want to override it...

File: src/common/components/Navbar/index.js (hunk 1 above)
Comment (Kamahl19) [reply src/common/components/Navbar/index.js]: Good point i wanted to do it but then forgot

File: src/features/auth/components/PageLayout/index.js (hunk 2 above)
Comment (Kamahl19) [reply src/features/auth/components/PageLayout/index.js]: I was thinking navbar could be an ant-based separate package or we could invlude it in ant codebase

File: src/features/auth/components/PageLayout/index.js (hunk 3 above)
Comment (Kamahl19) [reply src/features/auth/components/PageLayout/index.js]: Nope i will remove it altogether

**Comment by Kamahl19** [comment]
@MiroslavPetrik please review again

File: src/common/components/Navbar/index.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import PropTypes from 'prop-types';
+
+import ResponsiveNavigation from '../../../packages/responsive-navigation';
+
+import { Icon, Popover, Menu } from '../';
+
+const Navbar = ({ children, trigger }) => (
+  <ResponsiveNavigation>
+    {({ activePath, hideNavigation, isMobile, isNavigationVisible, showNavigation }) => {
+      if (children.type !== Navbar.Menu || !isValidElement(Children.only(children))) {
```
Comment (MiroslavPetrik) [inline src/common/components/Navbar/index.js]: Is there a value for which Children.only can pass and the result can be an invalid element?

Comment (Kamahl19) [reply src/common/components/Navbar/index.js]: thats a great question, React docs is not helpful here. Will dig in the React's source code

Comment (Kamahl19) [reply src/common/components/Navbar/index.js]: Ok so actually `Children.only` uses the `isValidElement` https://github.com/facebook/react/blob/b87aabdfe1b7461e7331abb3601d9e6bb27544bc/packages/react/src/ReactChildren.js#L387

**Pull request merged**